### PR TITLE
feat(sqlite): implement SQLite database schema and repository (Task 18)

### PR DESCRIPTION
**Pull request by TestardR**
## 🎯 Task 18: Design and Implement SQLite Database Schema

This PR implements a comprehensive SQLite database solution for the payment processing system, including schema design, migrations, connection pooling, and repository implementation.

### ✅ Database Schema Design
- **Payments table** with proper data types and constraints
- **Indexes** for efficient querying (idempotency_key, status, timestamps, IBANs)
- **Check constraints** for data validation (name lengths, positive amounts, valid statuses)
- **Unique constraints** for idempotency key enforcement
- **Automatic timestamp updates** via database triggers

### ✅ Migration System
- **Embedded SQL files** using Go's embed directive
- **Version tracking** with schema_migrations table
- **Idempotent migrations** that can be run multiple times safely
- **Migration status reporting** and validation
- **Checksum verification** for migration integrity

### ✅ Connection Pooling & Configuration
- **Configurable connection pool** with proper limits and timeouts
- **SQLite-specific optimizations** (WAL mode, busy timeout, cache size)
- **Health check functionality** for monitoring database status
- **Connection statistics** and monitoring support

### ✅ Repository Implementation
- **Full Repository interface** implementation for the Payment domain
- **Proper error handling** with domain-specific errors
- **Idempotency key enforcement** with duplicate detection
- **Status transitions** with validation
- **Concurrent operation support** with proper isolation

### ✅ Comprehensive Testing
- **Integration tests** for all database operations
- **Migration testing** with embedded files
- **Connection pooling tests** with concurrency validation
- **Error handling tests** for edge cases and failures
- **Transaction support** testing with rollback scenarios

### Database Schema
```sql
CREATE TABLE payments (
    id TEXT PRIMARY KEY NOT NULL,
    debtor_iban TEXT NOT NULL,
    debtor_name TEXT NOT NULL CHECK(length(debtor_name) >= 3 AND length(debtor_name) <= 30),
    creditor_iban TEXT NOT NULL,
    creditor_name TEXT NOT NULL CHECK(length(creditor_name) >= 3 AND length(creditor_name) <= 30),
    amount_cents INTEGER NOT NULL CHECK(amount_cents > 0),
    currency TEXT NOT NULL DEFAULT 'EUR',
    idempotency_key TEXT NOT NULL UNIQUE CHECK(length(idempotency_key) = 10),
    status TEXT NOT NULL CHECK(status IN ('PENDING', 'PROCESSED', 'FAILED')),
    created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
    updated_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP
);
```

### Key Features
- **Domain-driven design** with proper separation of concerns
- **Type safety** with Go domain objects and SQLite constraints
- **Performance optimized** with appropriate indexes and connection pooling
- **Error resilience** with proper error handling and validation
- **Test coverage** with comprehensive integration tests

## 🧪 Test Results
```
PASS
ok  	paymentprocessor/internal/infrastructure/persistence/sqlite	1.162s
```

All tests passing with comprehensive coverage of:
- Database initialization and migrations
- CRUD operations with proper validation
- Concurrent operations and race condition handling
- Error scenarios and edge cases
- Connection pooling and health checks

## 📋 Files Changed
- `internal/infrastructure/persistence/sqlite/` - Complete SQLite implementation
- `internal/domain/shared/errors.go` - Added duplicate idempotency key error
- `go.mod` - Added SQLite driver dependency

## 🔗 Related
- Implements **Task 18** from the project roadmap
- Foundation for **Task 20** (SQLite Repository) - already completed here
- Enables **Task 22** (Dual Persistence Transaction Manager)

Ready for review! 🚀

### CODE REVIEW

File: internal/infrastructure/persistence/sqlite/migrations/001_create_payments_table.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+-- Description: Create payments table with proper indexes and constraints
+-- Created: 2025-01-21
+
+-- Create payments table
+CREATE TABLE IF NOT EXISTS payments (
+    id TEXT PRIMARY KEY NOT NULL,
+    debtor_iban TEXT NOT NULL,
+    debtor_name TEXT NOT NULL CHECK(length(debtor_name) >= 3 AND length(debtor_name) <= 30),
+    creditor_iban TEXT NOT NULL,
+    creditor_name TEXT NOT NULL CHECK(length(creditor_name) >= 3 AND length(creditor_name) <= 30),
```
Comment (TestardR) [inline internal/infrastructure/persistence/sqlite/migrations/001_create_payments_table.sql:11]: drop length check, they are implemented at the Domain layer

File: internal/infrastructure/persistence/sqlite/migrations/001_create_payments_table.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    amount_cents INTEGER NOT NULL CHECK(amount_cents > 0),
+    currency TEXT NOT NULL DEFAULT 'EUR',
+    idempotency_key TEXT NOT NULL UNIQUE CHECK(length(idempotency_key) = 10),
```
Comment (TestardR) [inline internal/infrastructure/persistence/sqlite/migrations/001_create_payments_table.sql:14]: drop length check

File: internal/infrastructure/persistence/sqlite/database.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+// Database wraps sql.DB with additional functionality
+type Database struct {
+	db       *sql.DB
+	config   Config
+	migrator *Migrator
+}
+
+// NewDatabase creates a new database connection with the given configuration
+func NewDatabase(config Config) (*Database, error) {
```
Comment (TestardR) [inline internal/infrastructure/persistence/sqlite/database.go:61]: don't return pointer

File: internal/infrastructure/persistence/sqlite/database.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+		EnableWAL:         true,
+		EnableForeignKeys: true,
+	}
+}
+
+// Database wraps sql.DB with additional functionality
+type Database struct {
+	db       *sql.DB
+	config   Config
+	migrator *Migrator
```
Comment (TestardR) [inline internal/infrastructure/persistence/sqlite/database.go:57]: Migrator does not have to be a pointer

File: internal/infrastructure/persistence/sqlite/migrator.go [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+		Version: version,
+		Name:    name,
+		SQL:     string(sqlBytes),
+	}, nil
+}
+
+// getAppliedMigrations retrieves all applied migrations from the database
+func (m *Migrator) getAppliedMigrations(ctx context.Context) ([]Migration, error) {
+	query := `
+		SELECT version, name, checksum, applied_at
```
Comment (TestardR) [inline internal/infrastructure/persistence/sqlite/migrator.go:200]: drop name and checksum

File: internal/infrastructure/persistence/sqlite/payment_repository.go [hunk 6]
```diff
@@ -1,5 +1,216 @@
...
+)
+
+// PaymentRepository implements the payment.Repository interface using SQLite
 type PaymentRepository struct {
-	// TODO: Implement SQLite payment repository
+	db *Database
+}
+
+// NewPaymentRepository creates a new SQLite payment repository
+func NewPaymentRepository(db *Database) *PaymentRepository {
```
Comment (TestardR) [inline internal/infrastructure/persistence/sqlite/payment_repository.go:20]: don't return a pointer

File: internal/infrastructure/persistence/sqlite/payment_repository.go [hunk 7]
```diff
@@ -1,5 +1,202 @@
...
+	"fmt"
+	"time"
+
+	"paymentprocessor/internal/domain/payment"
+	"paymentprocessor/internal/domain/shared"
+)
+
 type PaymentRepository struct {
-	// TODO: Implement SQLite payment repository
+	db *Database
```
Comment (TestardR) [inline internal/infrastructure/persistence/sqlite/payment_repository.go:15]: Database should not be a pointer

File: internal/infrastructure/persistence/sqlite/payment_repository.go [hunk 8]
```diff
@@ -1,5 +1,202 @@
...
+		return fmt.Errorf("failed to save payment: %w", err)
+	}
+
+	return nil
+}
+
+func (r *PaymentRepository) FindByID(ctx context.Context, id string) (*payment.Payment, error) {
+	query := `
+		SELECT id, debtor_iban, debtor_name, creditor_iban, creditor_name,
+			   amount_cents, idempotency_key, status, created_at, updated_at
```
Comment (TestardR) [inline internal/infrastructure/persistence/sqlite/payment_repository.go:61]: on all methods, don't return a pointer, return a value for payment.Payment
